rs/nns/governance/src/neuron_store.rs

```diff
@@ -1390,11 +1390,26 @@ impl NeuronStore {
 pub fn prune_some_following(
     voting_power_economics: &VotingPowerEconomics,
     neuron_store: &mut NeuronStore,
-    mut next: Bound<NeuronId>,
-    mut carry_on: impl FnMut() -> bool,
+    next: Bound<NeuronId>,
+    carry_on: impl FnMut() -> bool,
 ) -> Bound<NeuronId> {
     let now_seconds = neuron_store.now();
+    groom_some_neurons(
+        neuron_store,
+        |neuron| {
+            neuron.prune_following(voting_power_economics, now_seconds);
+        },
+        next,
+        carry_on,
+    )
+}
 
+pub fn groom_some_neurons(
+    neuron_store: &mut NeuronStore,
+    mut touch_neuron: impl FnMut(&mut Neuron),
+    mut next: Bound<NeuronId>,
+    mut carry_on: impl FnMut() -> bool,
+) -> Bound<NeuronId> {
     loop {
```

Comment:
why not 

```suggestion
    while carry_on() { ... }
```

do you actually want the `do ... while` semantics here?

Reply:
NCR as this wasn't introduced in this PR.